File: android/app/src/main/kotlin/com/oasisforge/replybox/capture/ShippedApps.kt
package com.oasisforge.replybox.capture

/**
 * Decision 9, CAP-1, INB-20: the six apps captured before the user chooses
 * anything.
 *
 * A fourth copy of a list that already exists in lib/data/shipped_apps.dart, in
 * the manifest's `queries` element and on PERM-3's disclosure -- and it cannot be
 * avoided, because CAP-1's filter has to run before any Dart has ever executed on
 * this install. PERM-3 fails if any package can start enabled whose label the
 * disclosure never showed, so the copies have to be one set.
 *
 * This constant therefore lives alone, in its own file, in a shape a test outside
 * Kotlin can read without a parser: every double-quoted string in this file is a
 * package name and nothing else is quoted. Add a package here and to the three
 * other places at once, and keep any note about it comment text with no quotes in
 * it -- a quoted word anywhere in this file reads as a seventh package.
 */
object ShippedApps {

    /** Package names, exact. One per line; nothing else in this file is quoted. */
    val PACKAGES: Set<String> = setOf(
        "com.whatsapp",
        "com.facebook.orca", // Messenger
        "com.instagram.android",
        "org.telegram.messenger",
        "org.thoughtcrime.securesms", // Signal
        "com.google.android.apps.messaging", // Google Messages
    )
}

File: lib/data/shipped_apps.dart
/// The messaging apps Replybox captures from before the user chooses anything
/// (decision 9, CAP-1, INB-20).
///
/// Six, and exactly the six the spike's check 1 is defined over, so no
/// default-on app rests on behaviour nobody will verify on hardware. Every one
/// of them has to be listed in full on the permission disclosure (PERM-3) at
/// 1.3x text, so the length of this list is both the size of the privacy cost
/// and a layout constraint — it is not a place to be generous.
///
/// This is the single source that CAP-1's filter, INB-21's chooser, PERM-3's
/// disclosure and the manifest's `<queries>` all read. A test asserts this
/// constant and the manifest entries are one set, so the app can never
/// pre-enable a package whose label and icon the build cannot resolve.
library;

/// Package names, exact. Adding one here does **not** switch it on for an
/// install that already exists (decision 9): that user's bargain was struck at
/// their first run, and enabling an app they never saw would be a grant they
/// never gave (PERM-5).
const List<String> shippedMessagingApps = <String>[
  'com.whatsapp',
  'com.facebook.orca', // Messenger
  'com.instagram.android',
  'org.telegram.messenger',
  'org.thoughtcrime.securesms', // Signal
  'com.google.android.apps.messaging', // Google Messages
];

/// Whether a package is captured by default on a fresh install (CAP-1).
bool isShippedMessagingApp(String package) =>
    shippedMessagingApps.contains(package);
